pulsar/internal/commands.go

```diff
@@ -163,24 +164,26 @@ func ParseMessage(headersAndPayload []byte) (msgMeta *pb.MessageMetadata, payloa
 		return nil, nil, err
 	}
 
-    batchLen := make([]byte, 2)
-    if _, err = io.ReadFull(lr, batchLen); err != nil {
+    // Anything left in the frame is considered
+    // the payload and can be any sequence of bytes.
+    payloads := make([]byte, lr.N)
+    if _, err = io.ReadFull(lr, payloads); err != nil {
         return nil, nil, err
     }
 
-    // Anything left in the frame is considered
-    // the payload and can be any sequence of bytes.
-	if lr.N > 0 {
-		// guard against allocating large buffer
-		if lr.N > MaxFrameSize {
-			return nil, nil, fmt.Errorf("frame payload size (%d) "+
-					"cannot be greater than max frame size (%d)", lr.N, MaxFrameSize)
-		}
-		payload = make([]byte, lr.N)
-		if _, err = io.ReadFull(lr, payload); err != nil {
-			return nil, nil, err
-		}
-	}
+    numMsg := msgMeta.GetNumMessagesInBatch()
```

Comment:
the indent from line 167 to line 186 seems to be not aligned with line 188 and forward. is that an issue?

Reply:
Yes, will run `gofmt` to check code format.